### PR TITLE
Remove parentheses around multiple exception types in `except` and `except*` without `as`. 

### PR DESCRIPTION
**Pull request by ranjodhsingh1729**
### Description

Closes: #4678

Previously, catching multiple exception types required writing:

```python
try:
    …
except (A, B, C):
    …
```

With PEP 758 you can now omit the parentheses when you’re not using an `as` clause:

```python
try:
    …
except A, B, C:
    …
```

This change streamlines the syntax, reduces visual clutter when catching several exceptions, and brings `except` closer in style to other comma‑separated constructs in the language.

This PR implements Black’s support to remove parenthesis around multiple exception types in except and except* blocks when targeting Python 3.14 and above, in accordance with PEP 758.

### Checklist - did you ...

- [x] Implement any code style changes under the `--preview` style, following the
      stability policy?
- [x] Add an entry in `CHANGES.md` if necessary?
- [x] Add / update tests if necessary?
- [ ] Add new / update outdated documentation?

## **Additional Information**

* The `Grammer.txt` file has been modified.
* I haven't yet found a clean way to restrict the broader changes to only target Python 3.14+.

Any pointers on these points would be greatly appreciated. :)

### CODE REVIEW

**Comment by ranjodhsingh1729** [comment]
I have a question—hopefully one that’s easy enough to answer.

I understand that Black doesn’t need a grammar as detailed as what constructing an AST would require—just enough to reformat code. So, if it reformats valid syntax that only works in Python version x or higher, and the output doesn’t run on versions lower than x, that’s generally not an issue. It’s the user’s responsibility to ensure their code is compatible with the versions they target.

However, when Black starts formatting into newly introduced syntax, it should only do so when it's confident—either via heuristics or when explicitly told through --target-version. Otherwise, it risks breaking user code, which would be, at best, extremely frustrating. I don’t even want to think about the worst-case scenarios.

Right now, I’m trying to find out if Black has made such a change before, and I’ll keep digging. If anyone can point me to an example, that would be a huge help—I can take it from there.

**Comment by tusharsadhwani** [comment]
There's a `Feature` class IIRC, which uses the target version data and stores which features should be allowed in the code. Then within the formatter, parts of the code will check which features are enabled to decide how to format the code.

**Comment by ranjodhsingh1729** [comment]
Thanks — that did the trick.
I ended up using `supports_feature(mode.target_versions, Features.UNPARENTHESIZED_EXCEPTION_TUPLES)` to check the version, and `Preview.remove_parens_around_except_tuples in mode` to check if the feature is enabled (preview mode).
It feels a bit ad hoc, happy to refactor if there’s a cleaner way.

About the failed checks, I think that's because `py314` is wrongfully being added to `target_versions` by `detect_target_versions` during initialization. I'll look for a fix.

**Comment by cobaltt7** [comment]
That is typically the way we would do something like this, yes. Thanks for working on it!

**Comment by cobaltt7** [comment]
That is typically the way we would do something like this, yes. Thanks for working on it!

File: tests/data/cases/remove_braces_around_except.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
+# flags: --preview --minimum-version=3.14
```
Comment (cobaltt7) [inline tests/data/cases/remove_braces_around_except.py]: Could we also have a test that also runs on --preview but with a lower minimum version to ensure no invalid code is created?

Comment (ranjodhsingh1729) [reply tests/data/cases/remove_braces_around_except.py]: Sure i'll add it by tomorrow.

Comment (ranjodhsingh1729) [reply tests/data/cases/remove_braces_around_except.py]: I've added a test, and it actually helped catch an issue — I suspect it's the same one causing the PyPy3.9 macOS build to fail, but I'll know for sure after pushing.

The problem is that when mode.target_versions is empty, supports_feature ends up evaluating to all([]), which returns True. That means supports_feature(empty_set, some_feature) incorrectly returns True, and Black removes parentheses from except types even on older Python versions. ~Modifying support_feature causes multiple tests to fail which seem to be relying this quirk so i just check before calling it.~ (Sorry it was something else)

Since the only new syntax in Python 3.14 is the PEP 758, the only reliable way to detect target as python3.14 is by looking for unparenthesized except expressions. I've built a mechanism to do that — it detects one unparenthesized except expression and when in preview mode it unparenthesizes all. Should i add this?

The alternative would be to skip detection entirely and only apply the change when --target-version is explicitly set to 3.14 or higher. (which i was trying out when i stumbled on #4721).

Comment (ranjodhsingh1729) [reply tests/data/cases/remove_braces_around_except.py]: All set! I've added the tests along with the detection mechanism. The previously failing checks—the diff shades and the PyPy3.9 macOS build—are now passing too.

**Review by cobaltt7 — APPROVED** [review]

**Comment by ranjodhsingh1729** [comment]
@tusharsadhwani @cobaltt7 
Thanks for all the help, this was a great first contribution experience :)
